rust/otap-dataflow/crates/otlp/src/fake_signal_receiver/config.rs

```diff
@@ -23,25 +23,52 @@ pub enum OTLPSignal {
 /// Configuration should take a scenario to play out
 #[derive(Clone, Deserialize, Serialize)]
 pub struct Config {
-    // A ordered list of steps defining various signals to emit
-    steps: Vec<ScenarioStep>,
+    // Configuration of the traffic to generate
+    traffic_config: TrafficConfig,
 
     resolved_registry: ResolvedRegistry,
 }
 
+/// Configuration to describe the traffic being sent
+#[derive(Clone, Deserialize, Serialize)]
+pub struct TrafficConfig {
+    #[serde(default = "default_messages_per_second")]
+    messages_per_second: usize,
+    #[serde(default = "default_load")]
+    metric_load: Option<LoadConfig>,
+    #[serde(default = "default_load")]
+    trace_load: Option<LoadConfig>,
+    #[serde(default = "default_load")]
+    log_load: Option<LoadConfig>,
+}
+
+/// Configuration to describe how large the data should be
+#[derive(Clone, Deserialize, Serialize)]
+pub struct LoadConfig {
+    // number of messages produced
+    #[serde(default = "default_message_count")]
+    messages_per_scope: usize,
+    // number of scope produced
+    #[serde(default = "default_scope_count")]
+    scopes_per_resource: usize,
+    // number of resources produced
+    #[serde(default = "default_resource_count")]
+    resources_per_request: usize,
```

Comment:
Instead of having absolute numbers, I suggest to use relative weights.

First level of relative weights to describe the proportion of number of metrics, logs, spans in the global budget which is represented by the `signals_per_second` in the `TrafficConfig`.

For the last level of configuration, I suggest to simplify for now and only support 1 resource, with 1 scope and the number of signals determined with the previous parameters.